gtsfm/scene_optimizer.py

```diff
@@ -230,6 +237,51 @@ def _filter_dict_keys(dict: Dict[Any, Any], ref_dict: Dict[Any,Any]) -> Dict[Any
         # return the entry with just the sfm result
         return output_graph[0]
 
+def align_gtsfm_data(gtsfm_data: GtsfmData) -> Delayed:
```

Comment:
so we'll need to apply this to the GT, and to the estimated pose graphs, jointly. I propose that we estimate the transformation using the ground truth, and then add a classmethod to GtsfmData that aligns given a Pose3 object. And then call that classmethod on the estimated GtsfmData and the GT GtsfmData

Reply:
I've made the changes to estimate the alignment transformation based on `ba_output`, and then apply that to `ba_input` as well using a classmethod in GtsfmData. I will apply this transformation to the GT camera poses in the next PR, since this one is becoming faily lengthy already.